src/pages/Thanks/ThanksPage.vue

```diff
@@ -301,46 +302,55 @@ export default {
 	},
 	apollo: {
 		preFetch(config, client, { cookieStore, route }) {
-			const currentRoute = route.value ?? route ?? {};
-			const transactionId = currentRoute.query?.kiva_transaction_id
-				? numeral(currentRoute.query.kiva_transaction_id).value()
-				: null;
-
-			// Check if transactionId is null, resolve the promise if missing
-			if (!transactionId) {
-				logFormatter(
-					'Thanks page preFetch skipped due to missing transaction_id.',
-					'warning',
-				);
-				return Promise.resolve();
-			}
-
-			return client.query({
-				query: thanksPageQuery,
-				variables: {
-					checkoutId: transactionId,
-					visitorId: cookieStore.get('uiv') || null,
+			return Promise.all([
+				client.query({ query: thanksPageQuery }),
+				client.query({ query: experimentAssignmentQuery, variables: { id: 'share_ask_copy' } }),
+			]).then(() => {
+				const currentRoute = route.value ?? route ?? {};
+				const transactionId = currentRoute.query?.kiva_transaction_id
+					? numeral(currentRoute.query.kiva_transaction_id).value()
+					: null;
+				// Check if transactionId is null, resolve the promise if missing
+				if (!transactionId) {
+					logFormatter(
+						'Thanks page preFetch skipped due to missing transaction_id.',
+						'warning',
+					);
+					return Promise.resolve();
 				}
-			}).then(({ data }) => {
-				// Get teamId from receipt
-				let teamId = null;
-				const receipt = data?.shop?.receipt ?? null;
-				const loans = getLoans(receipt);
-				teamId = getTeamId(loans);
-
-				const filters = {
-					teamId,
-				};
-				const limit = 1;
 
-				return Promise.all([
-					client.query({ query: experimentAssignmentQuery, variables: { id: 'share_ask_copy' } }),
-					teamId ? fetchGoals(client, limit, filters) : null,
-					fetchPostCheckoutAchievements(client, getLoanIds(loans)),
-				]);
+				return client.query({
+					query: thanksPageReceiptQuery,
```

Comment:
If we can move the transaction ID check above, can this query be combined in the `Promise.all` above?

Reply:
For this and comment above, to resolve promise on line 319 I think this is a good approach.

Reply:
Right, but we resolved before running the other queries before the refactor. Does that not work after the refactor?

Reply:
Ah gotcha, makes sense, thanks for the explanation!